python-avd/pyavd/api/ip_addressing/__init__.py

```diff
@@ -222,6 +222,9 @@ def router_id(self) -> str:
         Default pool is "loopback_ipv4_pool"
         Default offset from pool is `id + loopback_ipv4_offset`
         """
+        if self.shared_utils.use_router_general_for_router_id is True and self.shared_utils.router_general is not None:
+            return get(self.shared_utils.router_general, "router_id.ipv4")
+
```

Comment:
This is doing the opposite of what we want. The purpose is to _not_ configure a router_id under router bgp or router ospf at all.

Reply:
overlay peering is using loopback0 ip which is generated from router-id. How to tackle this overlay peering issue  when we are not generating router-id?

Reply:
you should still be generating router-id. You should just not _output_ it under each routing protocol, but instead only under the router general. So the `shared_utils.router_id` should not change at all.

Reply:
Fixed.